### ISSUE TITLE
[Feature Request] Automate tracking prompts for a less disruptive annotation workflow

### ISSUE THREAD

**Issue by sen-trenser** [OHIF/Viewers#4622]
### What feature or change would you like to see made?

Currently OHIF asks a lot of prompts for tracking in different scenarios. This would be confusing for a user to understand what is tracking, untracking etc. The current tracking workflow is [documented here](https://docs.ohif.org/user-guide/viewer/measurement-tracking/#tracked-vs-untracked-measurements). This is alleviated in v3.9 with the `appConfig.disableConfirmationPrompts` config. 

This feature request is for further simplifying the UX for the underlying tracking workflow and reducing the involvement by users for workflow decisions (assuming the user wants the tracking workflow to update automatically with what he does with annotation or SR reports). 
This proposal is to remove the below scenarios from a user perspective based on the same configuration.
- "Untracked" annotations (Except non-acquisition oriented viewports in MPR views). 
- Displaying locked measurements in SR preview viewport.

The existing tracking workflow could be kept under the `appConfig.disableConfirmationPrompts` config. 

The below is the attached diagram for the proposed workflow.. 

![Image](https://github.com/user-attachments/assets/43026563-76ef-45f9-8186-892fb9bb1035)

Please share your suggestions or any preferred alternative suggestions to implement this requirements.

### Why should we prioritize this feature?

This will provide an easier and better experience for the users. 
This will also allows users to track measurement changes and allows to save their changes when necessary.

This proposal is to discuss and align the future plan from OHIF and ours to provide a better experience for users. We ([Flywheel](https://flywheel.io/)) would also plan to contribute the proposed changes to Cornerstone3D (for measurement changes tracking) and OHIF for the workflow changes once the proposal is approved. 

Could you please prioritize this?

**Comment by sedghi** [comment]
I think it makes sense to have three options for tracking, the "standard" (current), the "simplified" and "none" for disable, so we get rid of the `appConfig.disableConfirmationPrompts` and just have `appConfig.measurementTrackingMode` 

Is this workflow something you have iterated on based on your user interviews?

**Comment by kgoettler** [comment]
@sedghi to confirm my understanding:

- `standard`: current tracking workflow
- `simplified`: disable all prompts, except those mentioned by @sen-trenser above
- `none`: no tracking whatsoever (all measurements will be untracked)

Do I have that correct? 

> Is this workflow something you have iterated on based on your user interviews?

Yes. We've heard from users that they expect tracked measurements to be the default behavior, as they have little need for untracked measurements. The solution we've described here is the most straightforward way we've found to accomplish this without having to remove the measurement tracking functionality entirely (though if there was a way to do this while preserving the ability to save/export measurements as SR, we'd be interested in exploring that).

**Comment by sedghi** [comment]
@james-hanks

**Comment by james-hanks** [comment]
Agreed to review PR for config level change to allow simplified measurement tracking. Will review user data for future measurement tracking iteration

**Comment by sen-trenser** [comment]
@james-hanks @sedghi  We require Cornerstone modifications to track modified annotations with a dirty flag. Can we upgrade Cornerstone3D with the OHIF PR for these changes to enable this workflow? 

We would like to understand if there are no known compatibility issues in the latest version of CS3D that would prevent us from using it with OHIF 3.10. We would greatly appreciate any insights or advice you can provide on this matter.

What is the frequency of Cornerstone package upgrades in OHIF?

Thank you!

**Comment by sedghi** [comment]
Can you explain why cornerstone modifications are needed? What does the current flow do inside `cornerstone3D`, or it does not do it in cs3d

**Comment by sen-trenser** [comment]
@sedghi We are planning to have the isDirty flag as a state of cs3d annotation tools itself similar to [Lock/Visibility/Selection states](https://github.com/cornerstonejs/cornerstone3D/tree/main/packages/tools/src/stateManagement/annotation) of annotations. This can later be consumed by OHIF's measurement services to set or reset annotation dirty state based on annotation modification events, color change, label change or SR save workflow etc.

As `_calculateCachedStats()` of annotations trigger the `ANNOTATION_MODIFIED` event on first annotation render after loading an SR file, it would set the dirty flag on SR report loading/scrolling/jumpToMeasurement itself. So we would like to avoid using the `ANNOTATION_MODIFIED` event to set dirty flag. In order to identify the actual changes that affects the created SR report content, we would also like to add two events for `ANNOTATION_POINTS_MODIFIED` and `ANNOTATION_TEXT_BOX_MODIFIED` as well.

**Comment by sen-trenser** [comment]
Based on [the discussion in Office hours](https://github.com/OHIF/OHIF-Office-Hours/blob/main/notes/2025-01-30.md#-discuss-on-prs-related-to-annotation-tracking-simplification), the below path was suggested. 

> Use measurement service tracking instead of modifying Cornerstone.
> Check the annotation modified event with change types for better control.
> The [PR 4686](https://github.com/OHIF/Viewers/pull/4686) is WIP, that changes the tracking flow and handles multiple studies measurement display. Any changes to workflow if needed, should be done based on that PR.

On further analysis, the change type is not properly set in cs3d for annotation modified events.

The below list is the current change types used.

![Image](https://github.com/user-attachments/assets/65d4c71a-c68d-4f87-9efa-26cd2938dc20)

The below list would be an ideal* change type to implement. 

![Image](https://github.com/user-attachments/assets/c28065e9-e35d-4fcc-bb62-b168a3fffdb9)

* The StatsUpdated ChangeType triggered from renderAnnotation() would cause duplicate annotation modified events. This may also fire from scrolling after SR loading or scrolling after interpolation. If possible, this may need calculation at the time points are modified/SR is loaded, and then the modified event could be triggered.

@sedghi We are also waiting for our PO to have some suggestion for untracked annotation's behavior with annotations in multiple studies. 
Currently we are working on the cornerstone changes needed for correcting the ChangeTypes in annotation modified event. Is it okay to update the proposed changeTypes list partially in cs3d (to unblock this workflow) as given in above image?

**Comment by sedghi** [comment]
Let's do like only one tool say Length with proposed change and have a PR and we can look and discuss

**Comment by nithin-trenser** [comment]
@sedghi We have created a PR with proposed change in Length tool. Could you please have a look into it?
https://github.com/cornerstonejs/cornerstone3D/pull/1821
cc: @sen-trenser

**Comment by sedghi** [comment]
What's the status on this? We're ramping up tasks for 3.10, so if you want to include this, it really needs to be done by the end of the week, I think.

**Comment by nithin-trenser** [comment]
@sedghi 
Following the advice we received from the office hours meeting as [mentioned here](https://github.com/OHIF/Viewers/issues/4622#issuecomment-2651007653), we've created a [PR ](https://github.com/OHIF/Viewers/pull/4822)for this feature on the branch linked to PR https://github.com/OHIF/Viewers/pull/4712. All updates are complete, and the branch is now up to date with the parent branch. Just a heads up, the parent branch is still uses cornertone3d version 3.0.4, but we've upgraded to 3.1.0. We're hoping it will eventually be updated to 3.7.0 when it syncs with the master branch. We hope that our PR will also included in version 3.10. Please let us know if there is anything needed from our end to get this included in 3.10.

**Comment by nithin-trenser** [comment]
@sedghi 
We need to get a review on this. Since it includes all the updates from PR https://github.com/OHIF/Viewers/pull/4712 we need to merge changes from an updated parent branch([OHIF:feat/measurement-table-grouping](https://github.com/OHIF/Viewers/tree/feat/measurement-table-grouping)) into this PR branch.
Could you also please follow up with measurement-table-grouping to get it included in 3.10?
